File: leetcode/design/379_Design_Phone_Directory/codes/hash.py
class PhoneDirectory:
    def __init__(self, maxNumbers: int):
        self.slotsAvailable = set(range(maxNumbers))

    def get(self) -> int:
        if not self.slotsAvailable:
            return -1
        slot = next(iter(self.slotsAvailable))
        self.slotsAvailable.remove(slot)
        return slot

    def check(self, number: int) -> bool:
        return number in self.slotsAvailable

    def release(self, number: int) -> None:
        self.slotsAvailable.add(number)

# Your PhoneDirectory object will be instantiated and called as such:
# obj = PhoneDirectory(maxNumbers)
# param_1 = obj.get()
# param_2 = obj.check(number)
# obj.release(number)

File: leetcode/design/379_Design_Phone_Directory/codes/hash.cpp
#include <unordered_set>
using namespace std;

class PhoneDirectory {
    unordered_set<int> slotsAvailable;

public:
    PhoneDirectory(int maxNumbers) {
        for (int i = 0; i < maxNumbers; ++i)
            slotsAvailable.insert(i);
    }

    int get() {
        if (slotsAvailable.empty()) return -1;
        int slot = *slotsAvailable.begin();
        slotsAvailable.erase(slot);
        return slot;
    }

    bool check(int number) {
        return slotsAvailable.count(number) > 0;
    }

    void release(int number) {
        slotsAvailable.insert(number);
    }
};

/**
 * Your PhoneDirectory object will be instantiated and called as such:
 * PhoneDirectory* obj = new PhoneDirectory(maxNumbers);
 * int param_1 = obj->get();
 * bool param_2 = obj->check(number);
 * obj->release(number);
 */